### PR TITLE
CDRIVER-4044 consistent copyright date and license notice

### PR DESCRIPTION
**Pull request by eramongodb**
Resolves CDRIVER-4044.

The addition/removal of license notices in source files is deferred to CDRIVER-5613; this PR only ensures _current_ license notices are all made consistent.

Consistency is enforced such that the copyright template within the license notice follows this exact pattern:

```
Copyright 2009-present MongoDB, Inc.
```

where 2009 is the MongoDB C Driver project inception date and `MongoDB, Inc.` is the name of the owner.

The list of files was obtained using the following pattern matcher (`rg`: ripgrep):

```bash
rg --no-filename --ignore-case --replace '$1' '.*(copyright.*mongo[^"]*).*$' | sort -u
```

Which produced this initial list of patterns:

```
Copyright (C) 2013-present MongoDB, Inc.
Copyright 2009-2013 MongoDB, Inc.
Copyright 2009-present MongoDB, Inc.
Copyright 2013 MongoDB Inc.
Copyright 2013 MongoDB, Inc.
Copyright 2013-2014 MongoDB, Inc.
Copyright 2013-present MongoDB Inc.
Copyright 2013-present MongoDB, Inc.
Copyright 2014 MongoDB, Inc.
Copyright 2015 MongoDB Inc.
Copyright 2015 MongoDB, Inc.
Copyright 2015-present MongoDB, Inc.
Copyright 2016 MongoDB, Inc.
Copyright 2016-present MongoDB, Inc.
Copyright 2017 MongoDB Inc.
Copyright 2017 MongoDB, Inc.
Copyright 2017-present MongoDB, Inc.
Copyright 2018-present MongoDB Inc.
Copyright 2018-present MongoDB, Inc.
Copyright 2019 MongoDB, Inc.
Copyright 2019-present MongoDB, Inc.
Copyright 2020 MongoDB, Inc.
Copyright 2020-present MongoDB, Inc.
Copyright 2021 MongoDB, Inc.
Copyright 2021-present MongoDB, Inc.
Copyright 2022 MongoDB, Inc.
Copyright 2023 MongoDB, Inc.
Copyright 2023-present MongoDB, Inc.
Copyright 2024 MongoDB, Inc.
Copyright 2024-present MongoDB, Inc.
copyright = "2017-present, MongoDB, Inc
copyright = "2024, MongoDB Inc
copyright = u'2017, MongoDB, Inc'
```

Separating these into three distinct groups:

- `Copyright (C) ...`: in the `libbson.rc.in` and `libmongoc.rc.in` files.
- `Copyright <date><suffix> MongoDB, Inc.`: in most source files currently containing a license notice.
- `"<date><suffix>, MongoDB Inc"`: in `conf.py` files under `docs/dev/`, `src/libbson/doc/`, and `src/libmongoc/doc/`. It is unclear if these should be updated (copyright of docs vs. source?), so are currently left unchanged, although the missing comma was added regardless.
- `u'2017, MongoDB, Inc'`: in `conf.py` under `build/sphinx/homepage-config`. Similar to the above, currently left unchanged (copyright of docs vs. source?).

Question: should the copyright templates in `conf.py` files be updated as well to use `2009-present`?

### CODE REVIEW

File: src/kms-message/src/hexlify.c
Comment (kevinAlbs) [inline src/kms-message/src/hexlify.c:2]: Files in `src/kms-message` are copied from libmongocrypt. Suggest reverting changes in `src/kms-message`. Reverting may fix the [kms-divergence-check](https://spruce.mongodb.com/task/mongo_c_driver_smoke_kms_divergence_check_patch_e87940820c7c643172e7d005484e7cf1ee824bec_667c31d57a87820007728314_24_06_26_15_20_55?execution=0) task.

**Review by kevinAlbs — COMMENTED** [review]
> Question: should the copyright templates in `conf.py` files be updated as well to use `2009-present`?

I expect "yes" based on [this comment](https://jira.mongodb.org/browse/DRIVERS-409?focusedCommentId=1657731&focusedId=1657731&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1657731)

File: src/kms-message/src/hexlify.c
Comment (eramongodb) [reply src/kms-message/src/hexlify.c]: Done.

**Review by eramongodb — COMMENTED** [review]
Updated copyright strings in `conf.py` files. The comma after the date is preserved per [Sphinx documentation](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-copyright):
> A copyright statement in the style '2008, Author Name'.

File: docs/dev/conf.py
Comment (rcsanchez97) [inline docs/dev/conf.py:22]: Any particular reason for leaving the year as 2024?

File: src/libbson/libbson.rc.in
Comment (rcsanchez97) [inline src/libbson/libbson.rc.in:37]: Any reason for leaving the start year as 2013?

File: src/libmongoc/libmongoc.rc.in
Comment (rcsanchez97) [inline src/libmongoc/libmongoc.rc.in:37]: Any reason for leaving the start year as 2013?

**Review by rcsanchez97 — APPROVED** [review]
LGTM. I just made a few comments where it seemed like something might have been overlooked. Feel free to ignore if I wrongly flagged something that was intentional.

**Review by kevinAlbs — APPROVED** [review]
LGTM with pending comments addressed.

After merging, suggest creating a PR to the `debian/unstable` branch to apply the copyright update to the patch file: [0001_local_mathjax.diff](https://github.com/mongodb/mongo-c-driver/blob/debian/unstable/debian/patches/0001_local_mathjax.diff). I expect that may fix the `debian-package-build` task.

File: docs/dev/conf.py
Comment (eramongodb) [reply docs/dev/conf.py]: Good catch. Updated.

File: src/libbson/libbson.rc.in
Comment (eramongodb) [reply src/libbson/libbson.rc.in]: Good catch. I may have accidentally reverted the update to year in a local rebase. Fixed.

File: src/libmongoc/libmongoc.rc.in
Comment (eramongodb) [reply src/libmongoc/libmongoc.rc.in:37]: Good catch. I may have accidentally reverted the update to year in a local rebase. Fixed.

**Pull request merged**
